File: 102-binary-tree-level-order-traversal/102-binary-tree-level-order-traversal.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    fun levelOrder(root: TreeNode?): List<List<Int>> {
        root ?: return emptyList()
        val levels = mutableListOf<List<Int>>()
        var level = listOf(root)
        while (level.isNotEmpty()) {
            levels.add(level.map(TreeNode::`val`)) 
            level = level.flatMap { x -> listOf(x.left, x.right).filterNotNull() }
        }
        return levels
    }
}

File: 102-binary-tree-level-order-traversal/102-binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root):
        levels, level = [], [root] if root else []
        while level:
            levels.append([n.val for n in level])
            level = [c for n in level for c in [n.left, n.right] if c]
        return levels